### ISSUE TITLE
Slow pop-up scan

### ISSUE THREAD

**Issue by trunglebka** [xiaoyifang/goldendict-ng#816]
Firstly, thank you guys for improving a great project.

The problem I've encountered after installing the most recent master commit is that the pop-up scan is very slow. I haven't done any benchmark (by a watch?) yet but it is slow enough to make me try to install several master commits.

When running goldendict from terminal (built from latest master commit), this log was showed: "Unrecognized OpenGL version". I've tried "55fe56f6" commit and it has this problem too.

Eventually I must switch back to 77c919f7 to get acceptable delay time of scan-popup, and this build do not show above OpenGL log.

OS: Fedora 38
Nvidia driver: 530.41.03

**Comment by xiaoyifang** [comment]
need more information.

**Comment by trunglebka** [comment]
What kind of information do you need and how can I get it? 🤔

**Comment by shenlebantongying** [comment]
Are you using the popup by selection?

Between `Apr 30 -> May 21` there is this change https://github.com/xiaoyifang/goldendict-ng/pull/686

**Comment by trunglebka** [comment]
I'm usually use scan-popup by double click on the word (while Alt key being pressed)

**Comment by shenlebantongying** [comment]
Yes, I can confirm in your use-case, the slowdown is very obvious :sweat_smile: 

You can try enabling "Track clipboard change" and press Ctrl+C, if the popup shows instantly, then it is https://github.com/xiaoyifang/goldendict-ng/pull/686

Maybe we should expose this value? @trunglebka's use case may want absolutely no delays.

**Comment by xiaoyifang** [comment]
Does double click use this delay timer too?

**Comment by trunglebka** [comment]
Well, thank you @shenlebantongying, I've reduced the `selectionDelayTimer.setInterval` to 50 (ms?) and now it is work beautifully.

I think that if the config can be exposed in setting is extremely helpful. With me, popup just showing every selection is very annoying when we just want to copy or do something else. So, I choose to use key binding and it work flawlessly and I want it to show instantly.

**Comment by shenlebantongying** [comment]
> Does double click use this delay timer too?

"Double click" will cause a selection change event which uses the delay timer.
